risc0/zkvm/platform/src/syscall.rs

```diff
@@ -393,3 +404,49 @@ pub unsafe extern "C" fn sys_write(fd: u32, write_buf: *const u8, nbytes: usize)
         nbytes as u32,
     );
 }
+
+/// Retrieves the value of an environment variable, and stores as much
+/// of it as it can it in the memory at [out_words, out_words +
+/// out_nwords).  Returns the length of the value.
+///
+/// This is normally called twice to read an environment variable:
+/// Once to get the length of the value, and once to fill in allocated
+/// memory.
```

Comment:
Heads up, in the future if we integrate clippy, unsafe functions will warn with missing a `# Safety` block in the docstr, if you happen to have a rough idea of the lifetime assumptions of the args, putting them in a Safety block now would be a nice way to jump start that documentation.

reference: https://rust-lang.github.io/rust-clippy/rust-1.56.0/index.html#missing_safety_doc

Reply:
@flaub in https://github.com/risc0/risc0/pull/426#discussion_r1134625722_ indicated a preference for marking all syscalls as "unsafe", even if they don't write to memory or do anything else unsafe.

How do we want to reconcile that?   Should we just add a "# Safety" block saying "This is safe to call!  No worries here."?

Reply:
Hmmm good question, if there is not a straight forward answer now we could just defer until we decide to incorporate clippy.

Reply:
I think we should probably always have a safety comment that goes with any unsafe calls. The justification could be: "There are no known issues with calling this function" or something to that effect.